libraries/extractor/src/main/java/androidx/media3/extractor/ts/AdtsReader.java

```diff
@@ -527,7 +557,76 @@ private void parseAdtsHeader() throws ParserException {
       sampleSize -= CRC_SIZE;
     }
 
-    setReadingSampleState(output, sampleDurationUs, 0, sampleSize);
+    if (!hasOutputFormat && sampleSize >= AAC_PCE_MIN_SIZE) {
+      // As sample can fit a PCE, try reading it.
+      setReadingAacPceState(output, sampleDurationUs, sampleSize);
+    } else {
+      setReadingSampleState(output, sampleDurationUs, 0, sampleSize);
+    }
+  }
+
+  @RequiresNonNull({"pendingOutputFormat", "pceBuffer"})
+  void readAacProgramConfigElement() {
```

Comment:
Would be nice to reference the spec doc here to make it easier for future readers to see where this logic is defined. I guess it is coming from `ISO 14496-3 Audio (2005) Table 4.2`.

Reply:
Of course. I'll add a comment to the code. It's `ISO 13818-7 Advanced Audio Coding (2006) Table 25`. Yours works too, though it's from MPEG-4 instead of MPEG-2.